## Data Collection & Organization

Data has been manually downloaded from [this GitHub repo](https://github.com/fcakyon/instafake-dataset) and organized in the data folder. For this project, we don't need to build or run any data pipeline. Data and files will be added to this [Fake Instagram Account Detection repo](https://github.com/midori256/Fake-Instagram-Account-Detection).

## Data Definition

Goal: Gain an understanding of our data features to inform the next steps of our project.

In [2]:
# install libraries
import pandas as pd
import numpy as np

In [3]:
# load dataframes from json files
fake_df = pd.read_json("data/fakeAccountData.json")
real_df = pd.read_json("data/realAccountData.json")

In [4]:
real_df.head()

userFollowerCount  userFollowingCount  userBiographyLength  userMediaCount  \
0                258                 238                    0               0   
1                263                 482                   30              29   
2                 51                  78                    9               0   
3                297                 480                   22              25   
4                113                 242                    0              95   

   userHasProfilPic  userIsPrivate  usernameDigitCount  usernameLength  isFake  
0                 1              0                   0              10       0  
1                 1              1                   0               8       0  
2                 1              1                   0              10       0  
3                 1              1                   2               9       0  
4                 1              1                   0              10       0

In [5]:
fake_df.head()

userFollowerCount  userFollowingCount  userBiographyLength  userMediaCount  \
0                 25                1937                    0               0   
1                324                4122                    0               0   
2                 15                 399                    0               0   
3                 14                 107                    0               1   
4                264                4651                    0               0   

   userHasProfilPic  userIsPrivate  usernameDigitCount  usernameLength  isFake  
0                 1              1                   0              10       1  
1                 1              0                   4              15       1  
2                 0              0                   3              12       1  
3                 1              0                   1              10       1  
4                 1              0                   0              14       1

In [6]:
# check columns
# both dataframes have the same columns
real_df.columns

Index(['userFollowerCount', 'userFollowingCount', 'userBiographyLength',
       'userMediaCount', 'userHasProfilPic', 'userIsPrivate',
       'usernameDigitCount', 'usernameLength', 'isFake'],
      dtype='object')

### Column Descriptions

There are 9 columns which are defined as below:

1. user_media_count - Total number of posts, an account has.
2. user_follower_count - Total number of followers, an account has.
3. user_following_count - Total number of followings, an account has.
4. user_has_profil_pic - Whether an account has a profile picture, or not.
5. user_is_private - Whether an account is a private profile, or not.
6. user_biography_length - Number of characters present in account biography.
7. username_length - Number of characters present in account username.
8. username_digit_count - Number of digits present in account username.
9. is_fake - True, if account is a spam/fake account, False otherwise


In [9]:
# Check types of columns in real_df
real_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 994 entries, 0 to 993
Data columns (total 9 columns):
 #   Column               Non-Null Count  Dtype
---  ------               --------------  -----
 0   userFollowerCount    994 non-null    int64
 1   userFollowingCount   994 non-null    int64
 2   userBiographyLength  994 non-null    int64
 3   userMediaCount       994 non-null    int64
 4   userHasProfilPic     994 non-null    int64
 5   userIsPrivate        994 non-null    int64
 6   usernameDigitCount   994 non-null    int64
 7   usernameLength       994 non-null    int64
 8   isFake               994 non-null    int64
dtypes: int64(9)
memory usage: 70.0 KB


In [10]:
# Check types of columns in fake_df
fake_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 9 columns):
 #   Column               Non-Null Count  Dtype
---  ------               --------------  -----
 0   userFollowerCount    200 non-null    int64
 1   userFollowingCount   200 non-null    int64
 2   userBiographyLength  200 non-null    int64
 3   userMediaCount       200 non-null    int64
 4   userHasProfilPic     200 non-null    int64
 5   userIsPrivate        200 non-null    int64
 6   usernameDigitCount   200 non-null    int64
 7   usernameLength       200 non-null    int64
 8   isFake               200 non-null    int64
dtypes: int64(9)
memory usage: 14.2 KB


There are 994 records for real account, and only 200 records for fake accounts.

In [11]:
# check statistics for each dataframe
real_df.describe()

userFollowerCount  userFollowingCount  userBiographyLength  \
count         994.000000          994.000000           994.000000   
mean          419.891348          516.138833            25.034205   
std           366.998029          517.709885            34.128111   
min             1.000000            4.000000             0.000000   
25%           218.000000          267.000000             0.000000   
50%           345.000000          419.500000            12.000000   
75%           515.750000          614.000000            36.000000   
max          4492.000000         6640.000000           150.000000   

       userMediaCount  userHasProfilPic  userIsPrivate  usernameDigitCount  \
count      994.000000        994.000000     994.000000          994.000000   
mean        68.473843          0.986922       0.724346            0.266600   
std        113.963572          0.113668       0.447068            0.851721   
min          0.000000          0.000000       0.000000            0.000000   
25%          8.000000          1.000000       0.000000            0.000000   
50%         30.000000          1.000000       1.000000            0.000000   
75%         78.750000          1.000000       1.000000            0.000000   
max       1058.000000          1.000000       1.000000            7.000000   

       usernameLength  isFake  
count      994.000000   994.0  
mean        11.070423     0.0  
std          2.877679     0.0  
min          5.000000     0.0  
25%          9.000000     0.0  
50%         11.000000     0.0  
75%         13.000000     0.0  
max         22.000000     0.0

In [12]:
fake_df.describe()

userFollowerCount  userFollowingCount  userBiographyLength  \
count         200.000000          200.000000           200.000000   
mean          116.640000         1878.030000            11.980000   
std           289.906744         1871.377801            27.757558   
min             0.000000            0.000000             0.000000   
25%            10.750000          278.000000             0.000000   
50%            29.500000         1446.500000             0.000000   
75%           102.500000         2505.500000             4.500000   
max          3208.000000         7497.000000           138.000000   

       userMediaCount  userHasProfilPic  userIsPrivate  usernameDigitCount  \
count      200.000000        200.000000      200.00000          200.000000   
mean         3.535000          0.605000        0.32500            1.635000   
std         28.585036          0.490077        0.46955            1.902597   
min          0.000000          0.000000        0.00000            0.000000   
25%          0.000000          0.000000        0.00000            0.000000   
50%          0.000000          1.000000        0.00000            1.000000   
75%          1.000000          1.000000        1.00000            3.000000   
max        396.000000          1.000000        1.00000           10.000000   

       usernameLength  isFake  
count      200.000000   200.0  
mean        11.390000     1.0  
std          3.532747     0.0  
min          5.000000     1.0  
25%          9.000000     1.0  
50%         11.000000     1.0  
75%         13.000000     1.0  
max         30.000000     1.0

## Data Cleaning